mvpa2/cmdline/cmd_ttest.py

```diff
@@ -91,5 +118,13 @@ def run(args):
     s[out_of_mask] = 0
 
     verbose(1, "Saving to %s" % args.output)
-    nib.Nifti1Image(s, None, header=nis[0].header).to_filename(args.output)
+    filetype_out = guess_backend(args.output)
+    if filetype_out == 'nifti' and filetype == 'nifti':
+        nib.Nifti1Image(s, None, header=nis[0].header).to_filename(args.output)
+    else:
+        if filetype_out == 'nifti':
+            warning('Coercing output to hdf5 because input was hdf5')
+            args.output += '.hdf5'
```

Comment:
actually it might be quite cool if you could 
- check if there is a ds.a.mapper in one of the original datasets
- try to apply it map2nifti on result -- if succeeds -- no need to coerce, but can output nifti, if not -- better puke stating that requested file output type can't be obtained